gtk: convert Window (and some related files) to zig-gobject

jcollie opened the pull request.

jcollie committed:
gtk: convert Window (and some related files) to zig-gobject

pluiedev commented on src/apprt/gtk/Window.zig:
```diff
@@ -473,18 +505,20 @@ pub fn syncAppearance(self: *Window) !void {
-        c.adw_toolbar_view_set_top_bar_style(toolbar_view, toolbar_style);
-        c.adw_toolbar_view_set_bottom_bar_style(toolbar_view, toolbar_style);
+        toolbar_view: {
+            const toolbar_view = gobject.ext.cast(adw.ToolbarView, tab_overview.getChild() orelse break :toolbar_view) orelse break :toolbar_view;
```
That is *way* too long... Add a separate variable

pluiedev commented on src/apprt/gtk/winproto/wayland.zig:
```diff
@@ -3,12 +3,13 @@ const std = @import("std");
-const gtk = @import("gtk");
-const gtk4_layer_shell = @import("gtk4-layer-shell");
 const gdk = @import("gdk");
+const gdk_wayland = @import("gdk_wayland");
```
If we're using gdk_{wayland,x11} modules then I think we can drop their headers in `c.zig`

jcollie replied:
Fixed.

pluiedev commented on src/apprt/gtk/winproto/x11.zig:
```diff
@@ -4,9 +4,17 @@ const builtin = @import("builtin");
+const xlib = @import("xlib");
 
+// This needs to remain because of the legacy X11 API calls
 const c = @import("../c.zig").c;
```
I wonder if we can just directly use `@cImport` here
```suggestion
const c = @cImport({
  @cInclude("X11/Xlib.h");
  @cInclude("X11/Xatom.h");
});
```

jcollie replied:
Fixed.

pluiedev reviewed (changes requested):
Mostly looks good

jcollie committed:
gtk: address review comments

1. Remove usage of C header imports for gtk x11/wayland.
2. Move X11 C header imports to winproto_x11.zig
3. Clean up long line by breaking it up into multiple steps.
